src/hypothesis/internal/renaming.py

```diff
@@ -39,6 +39,9 @@ def with_name_check(**kwargs):
                     kwargs[t] = kwargs.pop(k)
             return f(**kwargs)
 
+        if with_name_check.__doc__ is None:
+            with_name_check.__doc__ = ''
```

Comment:
Nit: It doesn't *really* matter, but it vaguely annoys me in this case that we'd end up with a docstring starting with leading newlines. 

Given that this happens because of throwing away docstrings, maybe it would be better to just not update `__doc__` at all in that case?

Reply:
I think that’s better, yeah.